_sources/iteration/Exercises.rst

```diff
@@ -0,0 +1,344 @@
+
+      - Take a closer look at the while loop and conditional.
+
+    - The loop will run infinitely.
+
+      + The value of ``i`` will always be greater than 2, resulting in an infinite loop.
+
+.. mchoice:: iteration_mcq_4
+    :practice: T
+
+    What is the output of the code below?
+
+    .. code-block:: cpp
+
+     int main() {
+       int j = 6;
+       int i = j + 4;
+       cout << "People ";
+       i = i % 2;
+       i = i - 1;
+       while (i < 3) {
+         cout << "really ";
+         if (i > 0) {
+           cout << "really ";
+         }
+         i += 2;
+       }
+       cout << "like Joe's Pizza." << endl;
+     }
+
+    - People really like Joe's Pizza.
+
+      - Take a closer look at the while loop condition.
+
+    - People really really really like Joe's Pizza.
+
+      + The code will print out three "really"s, two from evaluating the while loop condition and one from evaluating the if condition before printing out "like Joe's Pizza"
+
+    - People really really really really really like Joe's Pizza.
+
+      - Take a closer look at the while loop condition.
+
+    - The loop will run infinetly.
+
+      - Take a closer look at the while loop condition and reassignment of ``i``
+
+.. mchoice:: iteration_mcq_5
+    :practice: T
+
+    What is the output of the code below?
+
+    .. code-block:: cpp
+
+      int main() {
+        int n = 10;
+        // cout << "Da ";
+        cout << "na ";
+        while (n != 3) {
+          cout << "na ";
+          n--;
+        }
+        cout << "Batman!";
+      }
+
+    - na na na na na na na na Batman!
+
+      + The code prints out eight "na"s before printing out "Batman!"
+
+    - na na na na na na na Batman!
+
+      - Look over the code carefully. There are output statements before the while loop.
+
+    - Da na na na na na na na na Batman!
+
+      - Will "Da" ever be printed?
+
+    - It will result in an infinite loop.
+
+      - Since we repeatedly decrement ``n`` inside the while loop, it will eventually be equal to 3 and the while loop will terminate.
+
+.. mchoice:: iteration_mcq_6
+    :practice: T
+
+    What is the output of the code below?
+
+    .. code-block:: cpp
+
+     int main() {
+       int n = 10;
+       cout << "Da ";
+       cout << "na ";
+       while (n != 3) {
+         cout << "na ";
+       }
+       cout << "Batman!";
+     }
+
+    - Batman!
+
+      - Take a closer look at the while loop.
+
+    - Da Batman!
+
+      - Take a closer look at the while loop.
+
+    - Da na na na na na na na na Batman!
+
+      - Take a closer look at the while loop.
+
+    - It will result in an infinite loop.
+
+      + Since we never change the value of ``n``, 10 will never equal 3 so the code will run forever.
+
+.. mchoice:: iteration_mcq_7
+    :practice: T
+
+    What is the output of the code below?
+
+    .. code-block:: cpp
+
+     int main() {
+       for (int i = 1; i < 6; ++i) {
+         cout << x << "\t" << pow (x, 5) / pow (x, 3) << endl;
```

Comment:
We don't really cover `\t` or pow(), so I think it would be better to remove this question